### DISCUSSION TITLE
Get request Header count efficiently

### DISCUSSION THREAD

**Discussion by ddevienne** [uNetworking/uWebSockets#1680] [Q&A] [answered]
There's no accessor for how many headers are on the request.

And `HttpRequest::HeaderIterator` is only a forward iterator,
thus `std::distance()` must walk the `headers` which is O(N) instead of O(1).

Sure `N < 50`, still, the `HttpParser` knows how many headers it parsed,
so couldn't it tell the `HttpRequest`, and we'd have an accessor?

**Comment by uNetworkingAB** [comment, marked as answer]
I ran a benchmark with and without iterating over the headers. It's not measurable, doesn't move the needle.

So while technically true, it's not a problem

**Reply by ddevienne** [reply]
Thanks for taking the time to benchmark it.

I still think it wouldn't hurt to record in the request what the parser already knows.
But not doing anything is obviously cheaper :).
